massa-execution/src/worker.rs

```diff
@@ -255,106 +253,137 @@ impl ExecutionWorker {
     /// called when the blockclique changes
     fn blockclique_changed(
         &mut self,
-        mut blockclique: BlockHashMap<Block>,
-        mut finalized_blocks: BlockHashMap<Block>,
+        blockclique: BlockHashMap<Block>,
+        finalized_blocks: BlockHashMap<Block>,
     ) -> Result<(), ExecutionError> {
-        // filter out any already-finalized blocks
-        // this can happen for example on bootstrap because the consensus vs execution bootstrap snapshots may be out of sync
-        blockclique.retain(|_b_id, b| b.header.content.slot > self.last_final_slot);
-        finalized_blocks.retain(|_b_id, b| b.header.content.slot > self.last_final_slot);
-
         // stop the current VM execution and reset state to final
```

Comment:
`/// Revert the VM to its latest SCE-final state by clearing its active slot history.`

It would be even better to have a `Step 1` in the spec, link to the paragraph at the top of the function, and add `Step 1` to the comment, so that both the code and spec follow a clear number of steps which can be easily reviewed for correctness.

Reply:
done